WoodenUtilities/src/main/java/com/bedrocklegends/woodenutilities/utility/IUpgradable.java

```diff
@@ -0,0 +1,51 @@
+package com.bedrocklegends.woodenutilities.utility;
+
+import com.bedrocklegends.woodenutilities.config.CommonConfig;
+import com.bedrocklegends.woodenutilities.config.WoodenConfig;
+import com.bedrocklegends.woodenutilities.item.UpgradeItem;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraftforge.fml.RegistryObject;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public interface IUpgradable {
```

Comment:
Rather than an interface, can we use an abstract class since this has implemented methods?

Reply:
Yeah, that works too

Reply:
Done in 49748c5